src/odemis/driver/_mccdaq.py

```diff
@@ -0,0 +1,700 @@
+        host computer to re-enumerate the device.
+        '''
+        self.h.write([self.RESET, 0, 0, 0, 0, 0, 0, 0])
+
+    def SetTrigger(self, type, chan=0):
+        '''
+        This command configures the external trigger for analog input.  The
+        trigger may be configured to activate with either a logic rising
+        edge or falling edge input.  Once the trigger is received, the analog
+        input will proceed as configured.  The EXTTRIG option must be used
+        in the AInScan command to utilize this feature.
+
+         type:  the type of trigger  (0 = external trigger falling edge, 1 = external trigger rising edge)
+         chan:  the digial bit that receives the trigger signal(0 for USB-120LS, 0-3 for miniLAB 1008)
+        '''
+        self.h.write([self.SET_TRIGGER, type, chan, 0, 0, 0, 0, 0, 0])
+
+    def SetID(self, id):
+        '''
+        This command stores an identifier on the unit.  Values of 0-255 are valid.  Note that 0
+        is used to flag uninitialized units.
+
+        id: user id number (0-255)
+        '''
+
+        if id < 0 or id > 255:
+            print('SetID: id out of range')
+            return
+        try:
+            self.h.write([self.SET_ID, id, 0, 0, 0, 0, 0, 0])
+        except:
+            print('Error in writing id')
+
+    def GetID(self):
+        '''
+        This function retrieves the id number stored on the device.  Note that a value of 0
+        is used to flag an uninitialized device.
+        '''
+        try:
+            self.h.write([self.GET_ID, 0, 0, 0, 0, 0, 0, 0])
+        except:
+            print('Error in reading id')
+        try:
+            value = self.h.read(8, 100)
+        except:
+            print('Error in reading id, value =', value)
+        return (value[0])
+
+    def volts(self, gain, value):
+        '''
+        converts raw values to volts
+        '''
+        volt = 0.0
+        if gain == self.SE_10_00V:
+            volt = value * 10.0 / 0x3ff
+        elif gain == self.BP_20_00V:
+            volt = value * 20.0 / 0x7ff
+        elif gain == self.BP_10_00V:
+            volt = value * 10.0 / 0x7ff
+        elif gain == self.BP_5_00V:
+            volt = value * 5.0 / 0x7ff
+        elif gain == self.BP_4_00V:
+            volt = value * 4.0 / 0x7ff
+        elif gain == self.BP_2_50V:
+            volt = value * 2.5 / 0x7ff
+        elif gain == self.BP_2_00V:
+            volt = value * 2.0 / 0x7ff
+        elif gain == self.BP_1_25V:
+            volt = value * 1.25 / 0x7ff
+        elif gain == self.BP_1_00V:
+            volt = value * 1.0 / 0x7ff
+        else:
+            raise ValueError('volts: Unknown range.')
+
+        return volt
+
+
+class usb_1208LS(usb_1208):
+    def __init__(self, serial=None):
+        self.productID = 0x0007a  # USB-1208LS
+        usb_1208.__init__(self)
+        try:
+            self.h.open(0x09db, self.productID, serial)
+        except Exception:
+            raise HwError
```

Comment:
You should raise an *instance* of an exception. Also, it might be handy to know what caused the exception.
```
        except Exception as ex:
            raise HwError("Failed to find USB device {self.productID}, serial {serial}: {ex}")
```